unit-tests/source_tree_test.py

```diff
@@ -26,15 +30,15 @@ def run(self):
                 tracker.append(self._reader.filename)
 
         db = SqliteStateDatabase(tmp_path)
-        emap: Dict[str, Union[Type[Task], Type[Command]]] = {
-            '.foo': DummyTask
-        }
+        smap = PathMap([
+            (r'.*\.foo', DummyTask),
```

Comment:
Note that here `.foo` means "any character followed by foo". Is that what you wanted? Did you actually mean `\.foo` which means ".foo"?

Reply:
The pattern is passed to `re.match` so I think it has to match the entire path (in the new version) - so it's `.*` which should match absolutely anything followed by `\.foo` meaning it must end with extension `.foo` - functionally I think this is identical to the behaviour before since it used to only look at the extension